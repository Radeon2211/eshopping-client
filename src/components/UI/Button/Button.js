import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './Button.sc';
import LoadingOverlay from '../LoadingOverlay/LoadingOverlay';

export default function Button(props) {
  const { clicked, isLoading, disabled, children, ...rest } = props;

  const loadingOverlay = isLoading ? <LoadingOverlay loaderSize="small" disableText /> : null;

  return (
    <SC.Button {...rest} onClick={clicked} disabled={disabled || isLoading}>
      {children}
      {loadingOverlay}
    </SC.Button>
  );
}

Button.defaultProps = {
  $size: 'small',
  type: 'button',
  $color: 'blue',
  $filled: false,
  $stretch: false,
  isLoading: false,
  disabled: false,
  clicked: () => {},
};

Button.propTypes = {
  $size: PropTypes.string,
  type: PropTypes.string,
  $color: PropTypes.string,
  $filled: PropTypes.bool,
  $stretch: PropTypes.bool,
  isLoading: PropTypes.bool,
  disabled: PropTypes.bool,
  clicked: PropTypes.func,
  children: PropTypes.node.isRequired,
};
