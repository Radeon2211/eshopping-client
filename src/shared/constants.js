import * as Yup from 'yup';
import { getCountries } from 'country-fns';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productConditions = {
  NEW: 'new',
  USED: 'used',
  NOT_APPLICABLE: 'not_applicable',
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([productConditions.NEW, productConditions.USED, productConditions.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
};

export const listOfAreaCodes = getCountries().map(({ name, dial }) => {
  const finalValue = `+${dial} ${name.split('(')[0].trim()}`;
  return {
    value: dial,
    label: finalValue,
  };
});

export const listOfCountries = getCountries().map(({ name }) => {
  const finalValue = name.split('(')[0].trim();
  return {
    value: finalValue,
    label: finalValue,
  };
});

export const productPages = {
  ALL_PRODUCTS: 'ALL_PRODUCTS',
  MY_PRODUCTS: 'MY_PRODUCTS',
  USER_PRODUCTS: 'USER_PRODUCTS',
};

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const orderTypes = {
  PLACED_ORDERS: 'PLACED_ORDERS',
  SELL_HISTORY: 'SELL_HISTORY',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const modalTypes = {
  ABOUT_WEBSITE: 'ABOUT_WEBSITE',
  ADD_ADMIN: 'ADD_ADMIN',
  ADD_PRODUCT: 'ADD_PRODUCT',
  BUY_PRODUCTS: 'BUY_PRODUCTS',
  CART_ITEM_ADDED: 'CART_ITEM_ADDED',
  CHANGE_NAME: 'CHANGE_NAME',
  CHANGE_EMAIL: 'CHANGE_EMAIL',
  CHANGE_USERNAME: 'CHANGE_USERNAME',
  CHANGE_PHONE_NUMBER: 'CHANGE_PHONE_NUMBER',
  CHANGE_ADDRESS: 'CHANGE_ADDRESS',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
  CHANGE_CONTACTS: 'CHANGE_CONTACTS',
  CHANGE_DELIVERY_ADDRESS: 'CHANGE_DELIVERY_ADDRESS',
  CLEAR_CART: 'CLEAR_CART',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  EDIT_PRODUCT: 'EDIT_PRODUCT',
  LOGIN: 'LOGIN',
  REMOVE_ADMIN: 'REMOVE_ADMIN',
  RESET_PASSWORD: 'RESET_PASSWORD',
  PENDING_USER_INFO: 'PENDING_USER_INFO',
  SEND_VERIFICATION_LINK: 'SEND_VERIFICATION_LINK',
  SIGNUP: 'SIGNUP',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: `Price - ascending` },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: `Price - ascending` },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const updateCartActions = {
  INCREMENT: 'INCREMENT',
  DECREMENT: 'DECREMENT',
  NUMBER: 'NUMBER',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};

export const userStatuses = {
  ACTIVE: 'active',
  PENDING: 'pending',
};
