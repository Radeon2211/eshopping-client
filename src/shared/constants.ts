import * as Yup from 'yup';
import countryCodes from 'country-codes-list';
import { ProductCondition, ProductPhotoFieldValue } from './types/enums';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([ProductCondition.NEW, ProductCondition.USED, ProductCondition.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
  photo: Yup.mixed()
    .test('is-error', (value) => value !== ProductPhotoFieldValue.ERROR)
    .nullable(),
};

export const listOfAreaCodes = countryCodes.customArray({
  value: '{countryCallingCode}',
  label: '+{countryCallingCode} {countryNameEn}',
});

export const listOfCountries = countryCodes.customArray({
  value: '{countryNameEn}',
  label: '{countryNameEn}',
});

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: 'Price - ascending' },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: 'Price - ascending' },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig: ScrollToOptions = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};
