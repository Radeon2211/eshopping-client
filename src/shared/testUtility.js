// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
