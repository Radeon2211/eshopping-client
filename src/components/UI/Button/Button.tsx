import * as SC from './Button.sc';
import LoadingOverlay from '../LoadingOverlay/LoadingOverlay';
import { PropsWithChildren } from '../../../shared/types/types';

export interface ButtonProps {
  $size?: 'small' | 'big';
  type?: 'submit' | 'reset' | 'button';
  $color?: 'blue' | 'red';
  $filled?: boolean;
  $stretch?: boolean;
  isLoading?: boolean;
  disabled?: boolean;
  clicked?: () => void;
}

export default function Button(props: PropsWithChildren<ButtonProps>) {
  const {
    clicked = () => {},
    isLoading = false,
    disabled = false,
    type = 'button',
    $size = 'small',
    $color = 'blue',
    $stretch = false,
    $filled = false,
    children,
    ...rest
  } = props;

  const loadingOverlay = isLoading ? <LoadingOverlay loaderSize="small" disableText /> : null;

  return (
    <SC.Button
      onClick={clicked}
      disabled={disabled || isLoading}
      type={type}
      $size={$size}
      $color={$color}
      $stretch={$stretch}
      $filled={$filled}
      {...rest}
    >
      {children}
      {loadingOverlay}
    </SC.Button>
  );
}
