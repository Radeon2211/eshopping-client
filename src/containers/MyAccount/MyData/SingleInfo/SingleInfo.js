import React from 'react';
import PropTypes from 'prop-types';
import Button from '../../../../components/UI/Button/Button';
import Heading from '../../../../components/UI/Heading/Heading';
import { singleInfoNames } from '../../../../shared/constants';
import FlexWrapper from '../../../../components/UI/FlexWrapper';
import PlainText from '../../../../components/UI/PlainText';

export default function SingleInfo({ name, content, clickHandler }) {
  let contentNode = '';
  if (name === singleInfoNames.ADDRESS) {
    contentNode = (
      <FlexWrapper $direction="column" $spacing="1">
        {content.map((value, idx) => (
          <PlainText key={idx} $size="3" $wordBreak="break-all">
            {value}
          </PlainText>
        ))}
      </FlexWrapper>
    );
  } else if (name === singleInfoNames.CONTACTS) {
    contentNode = (
      <FlexWrapper $direction="column" $spacing="1">
        <PlainText $size="3" $wordBreak="break-all">
          Email:&nbsp;
          {content.email ? 'visible' : 'hidden'}
        </PlainText>
        <PlainText $size="3" $wordBreak="break-all">
          Phone number:&nbsp;
          {content.phone ? 'visible' : 'hidden'}
        </PlainText>
      </FlexWrapper>
    );
  } else {
    contentNode = (
      <PlainText $size="3" $wordBreak="break-all">
        {content}
      </PlainText>
    );
  }

  let button = null;
  if (clickHandler) {
    button = (
      <Button clicked={clickHandler} data-testid={`SingleInfo-${name}-btn`}>
        Change
      </Button>
    );
  }

  return (
    <FlexWrapper $direction="column" $spacing="2" $align="start" data-testid={`SingleInfo-${name}`}>
      <Heading $variant="h4">{name}</Heading>
      {contentNode}
      {button}
    </FlexWrapper>
  );
}

SingleInfo.defaultProps = {
  clickHandler: undefined,
};

SingleInfo.propTypes = {
  name: PropTypes.string.isRequired,
  content: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      email: PropTypes.bool.isRequired,
      phone: PropTypes.bool.isRequired,
    }),
    PropTypes.arrayOf(PropTypes.string),
  ]).isRequired,
  clickHandler: PropTypes.func,
};
