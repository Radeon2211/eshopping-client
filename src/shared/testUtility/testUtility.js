// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';
import { v4 as uuidv4 } from 'uuid';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../../store/reducers/rootReducer';
import { productConditions } from '../constants';

export const defaultDeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: 'active',
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const createHistoryPageNumber = (pageNumber = 1, push = jest.fn()) => ({
  listen: jest.fn(),
  location: { pathname: '/products', search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push,
});

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
  quantitySold = 0,
  buyerQuantity = 0,
  description = '',
  condition = productConditions.NEW,
) => ({
  _id: productId || uuidv4(),
  name,
  price,
  quantity,
  quantitySold,
  buyerQuantity,
  description,
  condition,
  photo,
  seller: {
    username: sellerUsername,
  },
  __v: 0,
  createdAt: '2021-02-10T19:10:38.872Z',
  updatedAt: '2021-02-10T19:10:38.872Z',
});

export const createCartItem = (
  sellerUsername = 'username',
  quantity = 1,
  productId,
  price = 2,
  name = 'product name',
  productQuantity = 2,
  photo = false,
) => {
  const product = createProductItem(
    productId || uuidv4(),
    sellerUsername,
    productQuantity,
    price,
    name,
    photo,
  );

  return {
    _id: uuidv4(),
    quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
) => ({
  _id: productId || uuidv4(),
  name,
  photo,
  price,
  quantity,
  seller: {
    username: sellerUsername,
  },
});

export const createOrder = (
  products = [createTransactionAndOrderProdItem()],
  orderId = uuidv4(),
  sellerUsername,
  buyerUsername,
  overallPrice,
  createdAt,
  sellerEmail,
  sellerPhone,
) => {
  const seller = sellerUsername
    ? {
        username: sellerUsername,
        email: sellerEmail,
        phone: sellerPhone,
      }
    : null;

  const buyer = buyerUsername
    ? {
        username: buyerUsername,
      }
    : null;

  return {
    _id: orderId,
    seller,
    buyer,
    products,
    overallPrice,
    deliveryAddress: defaultDeliveryAddress,
    createdAt,
    updatedAt: createdAt,
    __v: 0,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...defaultState.product,
      ...initialProduct,
    },
    ui: {
      ...defaultState.ui,
      ...initialUI,
    },
  };
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

export const checkReqMethodAndURL = (moxiosInstance, method, url, reqNumber = 0) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests[reqNumber - 1];
    return method === prevRequest.config.method && url === prevRequest.config.url;
  }
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState,
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};
