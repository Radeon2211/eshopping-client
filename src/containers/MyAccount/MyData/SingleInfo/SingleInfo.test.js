import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import SingleInfo from './SingleInfo';
import { checkProps } from '../../../../shared/testUtility';
import { singleInfoNames } from '../../../../shared/constants';
import theme from '../../../../styled/theme';
import Button from '../../../../components/UI/Button/Button';
import { UserDataValue } from '../../../../styled/components';

const setUp = (name, content, clickHandler) => {
  const props = {
    name,
    content,
    clickHandler,
  };
  return mount(
    <ThemeProvider theme={theme}>
      <SingleInfo {...props} />
    </ThemeProvider>,
  );
};

describe('<SingleInfo />', () => {
  describe('Check prop types', () => {
    it('Should NOT throw a warning', () => {
      const props = {
        name: singleInfoNames.NAME,
        content: 'test name',
        clickHandler: () => {},
      };
      expect(checkProps(SingleInfo, props)).toBeUndefined();
    });

    it('Should throw a warning', () => {
      expect(checkProps(SingleInfo, {})).not.toBe(null);
    });
  });

  describe(`Check if renders correctly`, () => {
    it('Should render name, one content and <Button /> and fire off a clickHandler when <Button /> is clicked', () => {
      const clickHandlerFn = jest.fn();
      const wrapper = setUp(singleInfoNames.NAME, 'test name', clickHandlerFn);
      expect(wrapper.find('[data-test="name"]').first().text()).toEqual(singleInfoNames.NAME);
      expect(wrapper.find(UserDataValue)).toHaveLength(1);
      expect(wrapper.find(UserDataValue).text()).toEqual('test name');
      const button = wrapper.find(Button);
      expect(button).toHaveLength(1);
      button.simulate('click');
      expect(clickHandlerFn).toBeCalled();
    });

    it('Should NOT render a <Button /> if no clickHandler is given', () => {
      const wrapper = setUp(singleInfoNames.NAME, 'test name');
      expect(wrapper.find(Button)).toHaveLength(0);
    });

    it('Should render address', () => {
      const address = ['Street 1', '00-000 City', 'Poland'];
      const wrapper = setUp(singleInfoNames.ADDRESS, address);
      expect(wrapper.find(UserDataValue).at(0).text()).toEqual(address[0]);
      expect(wrapper.find(UserDataValue).at(1).text()).toEqual(address[1]);
      expect(wrapper.find(UserDataValue).at(2).text()).toEqual(address[2]);
    });

    it('Should render contacts with all visible', () => {
      const wrapper = setUp(singleInfoNames.CONTACTS, { email: true, phone: true });
      expect(wrapper.find(UserDataValue).at(0).text()).toEqual('Email: visible');
      expect(wrapper.find(UserDataValue).at(1).text()).toEqual('Phone number: visible');
    });

    it('Should render contacts with all hidden', () => {
      const wrapper = setUp(singleInfoNames.CONTACTS, { email: false, phone: false });
      expect(wrapper.find(UserDataValue).at(0).text()).toEqual('Email: hidden');
      expect(wrapper.find(UserDataValue).at(1).text()).toEqual('Phone number: hidden');
    });
  });
});
