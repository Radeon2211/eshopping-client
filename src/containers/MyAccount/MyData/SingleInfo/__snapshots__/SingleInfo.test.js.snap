// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleInfo /> check how renders should render correctly for address and with button 1`] = `
<DocumentFragment>
  .c4 {
  box-shadow: 0 0 3px rgba(0, 0, 0, 0.03);
  border-radius: 1px;
  cursor: pointer;
  font-weight: 700;
  line-height: 1.2;
  letter-spacing: 2px;
  outline: none;
  position: relative;
  text-align: center;
  text-transform: uppercase;
  transition: all 0.1s;
  font-size: 1.4rem;
  padding: 0.8rem 1.2rem;
  background-color: transparent;
  border: 2px solid #1E8BC3;
  color: #1E8BC3;
}

.c4:hover {
  border-color: #22A7F0;
  color: #22A7F0;
}

.c3 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  word-break: break-all;
  font-size: 1.5rem;
}

.c1 {
  font-size: 1.7rem;
  line-height: 1;
  margin-top: 0;
  text-align: left;
}

.c0 {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  align-items: start;
}

.c0>*:not(:last-child) {
  margin-bottom: 1.2rem;
}

.c2 {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
}

.c2>*:not(:last-child) {
  margin-bottom: 0.8rem;
}

<div
    class="c0"
    data-testid="SingleInfo-Address"
  >
    <h4
      class="c1"
    >
      Address
    </h4>
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        Street 1
      </span>
      <span
        class="c3"
      >
        00-000 City
      </span>
      <span
        class="c3"
      >
        Poland
      </span>
    </div>
    <button
      class="c4"
      data-testid="SingleInfo-Address-btn"
      type="button"
    >
      Change
    </button>
  </div>
</DocumentFragment>
`;

exports[`<SingleInfo /> check how renders should render correctly for contacts (both hidden) and without button 1`] = `
<DocumentFragment>
  .c3 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  word-break: break-all;
  font-size: 1.5rem;
}

.c1 {
  font-size: 1.7rem;
  line-height: 1;
  margin-top: 0;
  text-align: left;
}

.c0 {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  align-items: start;
}

.c0>*:not(:last-child) {
  margin-bottom: 1.2rem;
}

.c2 {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
}

.c2>*:not(:last-child) {
  margin-bottom: 0.8rem;
}

<div
    class="c0"
    data-testid="SingleInfo-Contacts"
  >
    <h4
      class="c1"
    >
      Contacts
    </h4>
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        Email: hidden
      </span>
      <span
        class="c3"
      >
        Phone number: hidden
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<SingleInfo /> check how renders should render correctly for contacts (both visible) and with button 1`] = `
<DocumentFragment>
  .c4 {
  box-shadow: 0 0 3px rgba(0, 0, 0, 0.03);
  border-radius: 1px;
  cursor: pointer;
  font-weight: 700;
  line-height: 1.2;
  letter-spacing: 2px;
  outline: none;
  position: relative;
  text-align: center;
  text-transform: uppercase;
  transition: all 0.1s;
  font-size: 1.4rem;
  padding: 0.8rem 1.2rem;
  background-color: transparent;
  border: 2px solid #1E8BC3;
  color: #1E8BC3;
}

.c4:hover {
  border-color: #22A7F0;
  color: #22A7F0;
}

.c3 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  word-break: break-all;
  font-size: 1.5rem;
}

.c1 {
  font-size: 1.7rem;
  line-height: 1;
  margin-top: 0;
  text-align: left;
}

.c0 {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  align-items: start;
}

.c0>*:not(:last-child) {
  margin-bottom: 1.2rem;
}

.c2 {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
}

.c2>*:not(:last-child) {
  margin-bottom: 0.8rem;
}

<div
    class="c0"
    data-testid="SingleInfo-Contacts"
  >
    <h4
      class="c1"
    >
      Contacts
    </h4>
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        Email: visible
      </span>
      <span
        class="c3"
      >
        Phone number: visible
      </span>
    </div>
    <button
      class="c4"
      data-testid="SingleInfo-Contacts-btn"
      type="button"
    >
      Change
    </button>
  </div>
</DocumentFragment>
`;

exports[`<SingleInfo /> check how renders should render correctly for name and with button 1`] = `
<DocumentFragment>
  .c3 {
  box-shadow: 0 0 3px rgba(0, 0, 0, 0.03);
  border-radius: 1px;
  cursor: pointer;
  font-weight: 700;
  line-height: 1.2;
  letter-spacing: 2px;
  outline: none;
  position: relative;
  text-align: center;
  text-transform: uppercase;
  transition: all 0.1s;
  font-size: 1.4rem;
  padding: 0.8rem 1.2rem;
  background-color: transparent;
  border: 2px solid #1E8BC3;
  color: #1E8BC3;
}

.c3:hover {
  border-color: #22A7F0;
  color: #22A7F0;
}

.c2 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  word-break: break-all;
  font-size: 1.5rem;
}

.c1 {
  font-size: 1.7rem;
  line-height: 1;
  margin-top: 0;
  text-align: left;
}

.c0 {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  align-items: start;
}

.c0>*:not(:last-child) {
  margin-bottom: 1.2rem;
}

<div
    class="c0"
    data-testid="SingleInfo-Name"
  >
    <h4
      class="c1"
    >
      Name
    </h4>
    <span
      class="c2"
    >
      Ludwig von Mises
    </span>
    <button
      class="c3"
      data-testid="SingleInfo-Name-btn"
      type="button"
    >
      Change
    </button>
  </div>
</DocumentFragment>
`;
